overlord/snapstate/handlers.go

```diff
@@ -2714,7 +2714,7 @@ func (m *SnapManager) undoLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	if len(snapst.Sequence.Revisions) == 1 {
+	if oldCurrent.Unset() {
```

Comment:
this mean doing this only on firstInstall ? maybe is a good idea to call the name the predicate like that and use the var? I'm not entirely sure I understand the need for this change

Reply:
During a refresh where we only refresh components, we treat it a lot like `snap refresh --revision=n snapname`, where `n` is the current revision of the snap. In that case, this undo handler wouldn't do the right thing. We don't want to treat it as if the snap is completely removed, since it isn't.

Naming the predicate to `firstInstall` is a good idea, I'll do that.

Reply:
Added tests for this case here: 541066d190a48a327489edfd1a27b306300c9e77